File: app/src/main/java/com/udacity/stockhawk/ui/StockDetailActivity.java
package com.udacity.stockhawk.ui;

import android.content.Intent;
import android.database.Cursor;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import butterknife.BindView;
import butterknife.ButterKnife;
import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.components.Description;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.data.LineDataSet;
import com.udacity.stockhawk.R;
import timber.log.Timber;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static com.udacity.stockhawk.data.Contract.Quote;

public class StockDetailActivity extends AppCompatActivity {
    private static final String TAG = StockDetailActivity.class.toString();

    @BindView(R.id.chart)
    LineChart chart;

    String stock;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_stock_detail);

        ButterKnife.bind(this);

        Intent intent = getIntent();
        stock = intent.getExtras().getString("stock");

        setTitle(stock);

        Uri uri = Quote.makeUriForStock(stock);

        Cursor c = getContentResolver().query(uri, Quote.QUOTE_COLUMNS, null, null, null);
        c.moveToFirst();
        String history = c.getString(c.getColumnIndexOrThrow(Quote.COLUMN_HISTORY));
        Timber.d(TAG, history);

        if (history != null) {
            List<Entry> entries = new ArrayList<>();
            int i = 0;
            String[] historyData = history.split("\n");

            // The items are stored in the database in descending order by data
            // Grab only the most recent 30 items
            List<String> recentItems = Arrays.asList(historyData).subList(0, 29);

            Timber.d(recentItems.toString());

            for (String rawEntry : recentItems) {
                String[] entry = rawEntry.split(",");

                entries.add(new Entry(Float.valueOf(i++), Float.valueOf(entry[1])));

            }

            LineDataSet dataSet = new LineDataSet(entries, "Stock Price");

            LineData lineData = new LineData(dataSet);

            chart.setData(lineData);
            chart.setDescription(new Description());
            chart.setDragEnabled(true);
            chart.setScaleEnabled(true);
            chart.getXAxis().setDrawLabels(false);
            chart.invalidate();
        }
    }


}

File: app/src/main/java/com/udacity/stockhawk/ui/StockDetailActivity.kt
package com.udacity.stockhawk.ui

import android.content.Intent
import android.database.Cursor
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import butterknife.BindView
import butterknife.ButterKnife
import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.components.Description
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.data.LineData
import com.github.mikephil.charting.data.LineDataSet
import com.udacity.stockhawk.R
import timber.log.Timber

import java.util.ArrayList
import java.util.Arrays

import com.udacity.stockhawk.data.Contract.Quote

class StockDetailActivity : AppCompatActivity() {

    @BindView(R.id.chart)
    internal var chart: LineChart? = null

    lateinit var stock: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_stock_detail)

        ButterKnife.bind(this)

        val intent = intent
        stock = intent.extras.getString("stock")

        title = stock

        val uri = Quote.makeUriForStock(stock)

        val c = contentResolver.query(uri, Quote.QUOTE_COLUMNS, null, null, null)
        c!!.moveToFirst()
        val history = c.getString(c.getColumnIndexOrThrow(Quote.COLUMN_HISTORY))
        Timber.d(TAG, history)

        if (history != null) {
            val entries = ArrayList<Entry>()
            var i = 0
            val historyData = history.split("\n".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

            // The items are stored in the database in descending order by data
            // Grab only the most recent 30 items
            val recentItems = Arrays.asList(*historyData).subList(0, 29)

            Timber.d(recentItems.toString())

            for (rawEntry in recentItems) {
                val entry = rawEntry.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

                entries.add(Entry(java.lang.Float.valueOf(i++.toFloat())!!, java.lang.Float.valueOf(entry[1])!!))

            }

            val dataSet = LineDataSet(entries, "Stock Price")

            val lineData = LineData(dataSet)

            chart!!.data = lineData
            chart!!.description = Description()
            chart!!.isDragEnabled = true
            chart!!.setScaleEnabled(true)
            chart!!.xAxis.setDrawLabels(false)
            chart!!.invalidate()
        }
    }

    companion object {
        private val TAG = StockDetailActivity::class.java.toString()
    }


}
